packages/block-library/src/navigation/edit/index.js

```diff
@@ -96,16 +98,23 @@ function Navigation( {
 		icon = 'handle',
 	} = attributes;
 
-	const ref = attributes.ref;
+	// Older versions of the block used an ID based ref attribute.
+	// Allow for this to continue to be used.
+	const ref = attributes.slug || attributes.ref;
 
-	const setRef = ( postId ) => {
-		setAttributes( { ref: postId } );
+	const [ idRef, setIdRef ] = useState( attributes.ref );
+
+	const setRef = ( { id, slug } ) => {
+		setAttributes( { slug } );
+		setIdRef( id );
 	};
 
 	const recursionId = `navigationMenu/${ ref }`;
 	const hasAlreadyRendered = useHasRecursion( recursionId );
 	const { editEntityRecord } = useDispatch( coreStore );
 
+	const generatedSlug = useGeneratedSlug( clientId );
```

Comment:
We can simplify this by removing the generation of the slug and just allowing the REST API to provide the slug to us that comes from WP Core.